modules/setting/server.go

```diff
@@ -322,10 +322,15 @@ func loadServerFrom(rootCfg ConfigProvider) {
 	}
 	StaticRootPath = sec.Key("STATIC_ROOT_PATH").MustString(StaticRootPath)
 	StaticCacheTime = sec.Key("STATIC_CACHE_TIME").MustDuration(6 * time.Hour)
-	AppDataPath = sec.Key("APP_DATA_PATH").MustString(path.Join(AppWorkPath, "data"))
+	AppDataPath = sec.Key("APP_DATA_PATH").String()
+	if AppDataPath == "" {
+		AppDataPath = "data"
+	}
 	if !filepath.IsAbs(AppDataPath) {
-		log.Info("The provided APP_DATA_PATH: %s is not absolute - it will be made absolute against the work path: %s", AppDataPath, AppWorkPath)
 		AppDataPath = filepath.ToSlash(filepath.Join(AppWorkPath, AppDataPath))
+		if InstallLock {
+			log.Error("The provided APP_DATA_PATH is not absolute, you must set it to absolute like APP_DATA_PATH=%s", AppDataPath)
```

Comment:
This error message doesn't make sense in its new form.
`AppDataPath` is the component that didn't work, right?
Why should we set it like the invalid path?`

Reply:
Okay, just noticed the line above.